sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosTracerTest.java

```diff
@@ -0,0 +1,375 @@
+            .buildAsyncClient();
+        cosmosAsyncDatabase = getSharedCosmosDatabase(client);
+        cosmosAsyncContainer = getSharedMultiPartitionCosmosContainer(client);
+
+    }
+
+    @Test(groups = {"emulator"}, timeOut = TIMEOUT)
+    public void cosmosAsyncClient() {
+        Tracer mockTracer = Mockito.mock(Tracer.class);
+        TracerProvider tracerProvider = Mockito.spy(new TracerProvider(getMockTracer(mockTracer)));
+        ReflectionUtils.setTracerProvider(client, tracerProvider);
+
+        TracerProviderCapture tracerProviderCapture = new TracerProviderCapture();
+        Mockito.doAnswer(tracerProviderCapture).when(tracerProvider).startSpan(Matchers.anyString(),
+            Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+
+        client.createDatabaseIfNotExists(cosmosAsyncDatabase.getId()).block();
+        Context context = tracerProviderCapture.getResult();
+        Mockito.verify(tracerProvider, Mockito.times(1)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+        verifyTracerAttributes(mockTracer, "createDatabaseIfNotExists." + cosmosAsyncDatabase.getId(), context,
+            cosmosAsyncDatabase.getId(), 1);
+
+        client.readAllDatabases(new CosmosQueryRequestOptions()).byPage().single().block();
+        Mockito.verify(tracerProvider, Mockito.times(2)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+        verifyTracerAttributes(mockTracer, "readAllDatabases", context, null, 2);
+
+
+        String query = "select * from c where c.id = '" + cosmosAsyncDatabase.getId() + "'";
+        client.queryDatabases(query, new CosmosQueryRequestOptions()).byPage().single().block();
+        Mockito.verify(tracerProvider, Mockito.times(3)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+        verifyTracerAttributes(mockTracer, "queryDatabases", context, null, 3);
+    }
+
+    @Test(groups = {"emulator"}, timeOut = TIMEOUT)
+    public void cosmosAsyncDatabase() {
+        Tracer mockTracer = Mockito.mock(Tracer.class);
+        TracerProvider tracerProvider = Mockito.spy(new TracerProvider(getMockTracer(mockTracer)));
+        ReflectionUtils.setTracerProvider(client, tracerProvider);
+        TracerProviderCapture tracerProviderCapture = new TracerProviderCapture();
+        Mockito.doAnswer(tracerProviderCapture).when(tracerProvider).startSpan(Matchers.anyString(),
+            Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+
+        cosmosAsyncDatabase.createContainerIfNotExists(cosmosAsyncContainer.getId(),
+            "/pk", 5000).block();
+        Context context = tracerProviderCapture.getResult();
+        Mockito.verify(tracerProvider, Mockito.times(1)).startSpan(Matchers.anyString(), Matchers.anyString(),
```

Comment:
Suggestion can be updated to check for the span name here
```
Mockito.verify(tracerProvider, Mockito.times(1)).startSpan(eq("createContainerIfNotExists"), Matchers.anyString()
````

Reply:
Mockito does not work with partial matching, it will give runtime error , and anyway we are checking span name on mock tracer so this does not needed

Reply:
Not sure what you mean, an example of what I am suggesting https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubProducerAsyncClientTest.java#L426

Reply:
@simplynaveen20 - we should an incrementing counter instead of using numeral literal values in `Mockito.times(1)` call - so that it is much clear to us.

Reply:
added incremental variable instead of numeral literal